### PR TITLE
UIDATIMP-373: Correct Repeatable Sections :: WIP

### PR DESCRIPTION
**Pull request by taras-tkachenko**

### CODE REVIEW

File: src/components/FlexibleForm/Control.js [hunk 1]
```diff
@@ -62,25 +77,10 @@ export const Control = memo(props => {
...
-      };
-    });
-  };
-
-  const getActualName = name => {
-    const nsName = name && sectionNamespace ? name.split('**ns**').join(sectionNamespace) : name;
-
-    return nsName && repeatableIndex >= 0 ? nsName.split('##ri##').join(repeatableIndex) : nsName;
-  };
+  console.log('Children: ', children);
```
Comment (mariia-aloshyna) [inline src/components/FlexibleForm/Control.js]: Remove this `console.log`, please

File: src/components/FlexibleForm/Control.js [hunk 2]
```diff
@@ -256,6 +256,24 @@ export const Control = memo(props => {
       return <>&nbsp;</>;
     }
 
+    const {
+      childControls: cc,
```
Comment (mariia-aloshyna) [inline src/components/FlexibleForm/Control.js]: Why did you rename this const? In my opinion, `cc` is not a clear name for the variable.

File: src/components/Section/Section.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const Section = memo(({
   children,
 }) => {
   const [isChecked, setChecked] = useState(optional ? enabled : true);
-
+  // eslint-disable-next-line
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.js]: Which error is disabled here?

File: src/settings/MatchProfiles/ViewMatchProfile.js [hunk 4]
```diff
@@ -218,22 +218,22 @@ export class ViewMatchProfile extends Component {
...
       ...matchProfile,
-      matchDetails: [{
-        incomingMatchExpression: {
-          fields: [
-            {
+      matchDetails: matchProfile.matchDetails.map(item => {
+        const {
+          incomingMatchExpression,
+          existingMatchExpression,
+        } = item;
```
Comment (mariia-aloshyna) [inline src/settings/MatchProfiles/ViewMatchProfile.js]: Such variable name like 'item' is meaningless for me. Consider renaming it.

File: src/settings/MatchProfiles/ViewMatchProfile.js [hunk 5]
```diff
@@ -245,40 +245,46 @@ export class ViewMatchProfile extends Component {
...
       ? <FormattedMessage id={FOLIO_RECORD_TYPES[matchProfile.existingRecordType].captionId} />
       : '';
     const incomingRecordLabel = INCOMING_RECORD_TYPES[matchProfile.incomingRecordType]
       ? <FormattedMessage id={INCOMING_RECORD_TYPES[matchProfile.incomingRecordType].captionId} />
       : '';
     const componentsProps = {
+      'section-incoming-field': { stateFieldValue: matchProfile.incomingRecordType },
+      'section-incoming-qualifier': { stateFieldValue: matchProfile.incomingRecordType },
+      'section-incoming-qualifier-part': { stateFieldValue: matchProfile.incomingRecordType },
+      // 'section-incoming-static-value-text': { stateFieldValue: staticValueType },
```
Comment (mariia-aloshyna) [inline src/settings/MatchProfiles/ViewMatchProfile.js]: If this comment is unused, remove it, please.

File: src/components/FlexibleForm/Control.js [hunk 6]
```diff
@@ -29,9 +29,24 @@ const controls = {
...
+
+  return nsName && repeatableIndex >= 0 ? nsName.split('##ri##').join(repeatableIndex) : nsName;
+};
+const getOptions = (options, sectionNamespace, intl) => {
+  return options.map(option => {
+    const lbl = option.label;
+
+    return {
+      value: option.value,
+      label: intl.formatMessage({ id: lbl && sectionNamespace ? lbl.split('**ns**').join(sectionNamespace) : lbl }),
```
Comment (VictorSoroka1) [inline src/components/FlexibleForm/Control.js]: I see that 
```lbl && sectionNamespace ? lbl.split('**ns**').join(sectionNamespace) : lbl``` check pretty much the same as
```name && sectionNamespace ? name.split('**ns**').join(sectionNamespace) : name``` [here](https://github.com/folio-org/ui-data-import/pull/444/files#diff-6709362c6d184124da81079fe4cef458R33)

I think some util function would be helpful here.

File: src/components/FlexibleForm/Control.js (hunk 2 above)
Comment (VictorSoroka1) [reply src/components/FlexibleForm/Control.js]: The same [here](https://github.com/folio-org/ui-data-import/pull/444/files#diff-6709362c6d184124da81079fe4cef458R250). `ri` is not readable. Even in this PR it took time for me to understand what it means as the line with its assignment is collapsed.

File: src/settings/MatchProfiles/ViewMatchProfile.js (hunk 4 above)
Comment (taras-tkachenko) [reply src/settings/MatchProfiles/ViewMatchProfile.js]: It is a variable inside a cycle that represents iterated item

File: src/settings/MatchProfiles/ViewMatchProfile.js (hunk 5 above)
Comment (taras-tkachenko) [reply src/settings/MatchProfiles/ViewMatchProfile.js]: It will be removed later as a part of states handling fix but not now.
